src/zenml/stack_deployments/stack_deployment.py

```diff
@@ -0,0 +1,133 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+
+#       https://www.apache.org/licenses/LICENSE-2.0
+
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Functionality to deploy a ZenML stack to a cloud provider."""
+
+import datetime
+from abc import abstractmethod
+from typing import ClassVar, Dict, List, Optional, Tuple
+
+from pydantic import BaseModel
+
+from zenml.enums import StackDeploymentProvider
+from zenml.models import (
+    DeployedStack,
+)
+
+
+class ZenMLCloudStackDeployment(BaseModel):
+    """ZenML Cloud Stack CLI Deployment base class."""
+
+    provider: ClassVar[StackDeploymentProvider]
+    stack_name: str
+    location: Optional[str] = None
+
+    @classmethod
+    @abstractmethod
+    def description(cls) -> str:
```

Comment:
I am a bit curious about the approach here. Considering this is a `pydantic` model, why did you separate values like `description` or `instructions` as `abstractmethod`s?

Reply:
I may have started with it being a pydantic model, but that's irrelevant now. I might as well remove BaseModel as a base class.